src/esssans/conversions.py

```diff
@@ -99,27 +117,89 @@ def phi(cylindrical_x: sc.Variable, cylindrical_y: sc.Variable) -> sc.Variable:
     return sc.atan2(y=cylindrical_y, x=cylindrical_x)
 
 
-def sans_elastic(gravity: bool = False, scatter: bool = True) -> dict:
+ElasticCoordTransformGraph = NewType('ElasticCoordTransformGraph', dict)
+MonitorCoordTransformGraph = NewType('MonitorCoordTransformGraph', dict)
+
+
+def sans_elastic(gravity: Optional[CorrectForGravity]) -> ElasticCoordTransformGraph:
     """
     Generate a coordinate transformation graph for SANS elastic scattering.
     By default, the effects of gravity on the neutron flight paths are not included.
 
     :param gravity: Take into account the bending of the neutron flight paths from the
         Earth's gravitational field if ``True``.
     """
-    graph = {**beamline.beamline(scatter=scatter), **tof.elastic_Q('tof')}
+    graph = {**beamline.beamline(scatter=True), **tof.elastic_Q('tof')}
     if gravity:
         graph['two_theta'] = two_theta
     graph['cyl_x_unit_vector'] = cyl_x_unit_vector
     graph['cyl_y_unit_vector'] = cyl_y_unit_vector
     graph['cylindrical_x'] = cylindrical_x
     graph['cylindrical_y'] = cylindrical_y
     graph['phi'] = phi
-    return graph
+    return ElasticCoordTransformGraph(graph)
 
 
-def sans_monitor() -> dict:
+def sans_monitor() -> MonitorCoordTransformGraph:
     """
     Generate a coordinate transformation graph for SANS monitor (no scattering).
     """
-    return {**beamline.beamline(scatter=False), **tof.elastic_wavelength('tof')}
+    return MonitorCoordTransformGraph(
+        {**beamline.beamline(scatter=False), **tof.elastic_wavelength('tof')}
+    )
+
+
+def monitor_to_wavelength(
+    monitor: RawMonitor[RunType, MonitorType], graph: MonitorCoordTransformGraph
+) -> WavelengthMonitor[RunType, MonitorType]:
+    return WavelengthMonitor[RunType, MonitorType](
+        monitor.transform_coords('wavelength', graph=graph)
+    )
+
+
+# TODO This demonstrates a problem: Transforming to wavelength should be possible
+# for RawData, MaskedData, ... no reason to restrict necessarily.
+# Would we be fine with just choosing on option, or will this get in the way for users?
+def detector_to_wavelength(
+    detector: MaskedData[RunType],
```

Comment:
Currently, we do the masking in two steps: first mask pixels, then convert to wavelength, then apply a wavelength mask. Any reason why we couldn't do all the masking in the same step? (e.g. after the conversion to wavelength?)

Reply:
I think it makes sense to do masking "as early as possible" (as in the original workflow?). We may want to verify pixel masking before doing subsequent steps (such as converting to wavelength).